crates/cli/tests/cli_run.rs

```diff
@@ -1402,54 +1064,56 @@ mod cli_run {
 
         #[cfg(feature = "i386-cli-run")]
         fn check_output_i386(
-            file_name: &Path,
-            stdin: &[&str],
+            file_path: &std::path::Path,
+            stdin: Vec<&'static str>,
             expected_ending: &str,
-            use_valgrind: UseValgrind,
         ) {
-            use super::{concatcp, CMD_BUILD, TARGET_FLAG};
-
-            check_output_with_stdin(
-                &file_name,
-                stdin,
-                &[concatcp!(TARGET_FLAG, "=x86_32")],
-                &[],
-                &[],
-                expected_ending,
-                use_valgrind,
-                TestCliCommands::Run,
-            );
-
-            check_output_with_stdin(
-                &file_name,
-                stdin,
-                &[concatcp!(TARGET_FLAG, "=x86_32"), OPTIMIZE_FLAG],
-                &[],
-                &[],
-                expected_ending,
-                use_valgrind,
-                TestCliCommands::Run,
-            );
+            use super::{concatcp, TARGET_FLAG};
+
+            let i386_target_arg = concatcp!(TARGET_FLAG, "=x86_32");
+
+            let runner = Run::new_roc()
+                .arg(CMD_RUN)
+                .add_arg_if(LINKER_FLAG, TEST_LEGACY_LINKER)
+                .arg(i386_target_arg)
+                .arg(file_path)
+                .with_stdin_vals(stdin.clone());
+
+            let out = runner.run();
+            out.assert_clean_success();
+            out.assert_stdout_and_stderr_ends_with(expected_ending);
+
+            let run_optimized = Run::new_roc()
+                .arg(CMD_RUN)
+                .add_arg_if(LINKER_FLAG, TEST_LEGACY_LINKER)
+                .arg(i386_target_arg)
+                .arg(OPTIMIZE_FLAG)
+                .arg(file_path)
+                .with_stdin_vals(stdin.clone());
+
+            let out_optimized = run_optimized.run();
+            out_optimized.assert_clean_success();
+            out_optimized.assert_stdout_and_stderr_ends_with(expected_ending);
         }
 
         #[test]
         #[cfg_attr(windows, ignore)]
         fn nqueens() {
-            test_benchmark("nQueens.roc", &["6"], "4\n", UseValgrind::Yes)
+            test_benchmark("nQueens.roc", vec!["6"], "4\n", UseValgrind::Yes)
```

Comment:
Why does this use vec now?

Reply:
I think I added it to resolve a lifetime issue... but that is probably my lack of rust foo. If we can make it work passing a reference that would be better. At the time I figured I would just do this and move on with the changes, and we could circle back to it later.